File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/DTOs/KingOfTheHill/MatchStartedData.cs
using GeoQuiz_backend.Domain.Entities;

namespace GeoQuiz_backend.Application.DTOs.KingOfTheHill
{
    public class MatchStartedData
    {
        public Guid MatchId { get; set; }
        public int TotalPlayers { get; set; }
        public int TotalRounds { get; set; } 
        public DateTime FirstRoundStartTime { get; set; }
        public List<PlayerInfo> AllPlayers { get; set; } = new(); 
    }

    public class PlayerInfo
    {
        public Guid PlayerId { get; set; }
        public string PlayerName { get; set; } = null!;
        public int PlayerLevel { get; set; }
        public static PlayerInfo FromPlayerLobby(PlayerLobby player)
        {
            return new PlayerInfo
            {
                PlayerId = player.Id,
                PlayerName = player.Name,
                PlayerLevel = player.Level
            };
        }
    }
}

File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/koth/MatchStartedData.java
package com.example.geoquiz_frontend.data.remote.dtos.koth;

import java.util.List;

public class MatchStartedData {
    private String matchId;

    private int totalPlayers;

    private int totalRounds;

    private String firstRoundStartTime;

    private List<PlayerInfo> allPlayers;

    public String getMatchId() { return matchId; }
    public int getTotalPlayers() { return totalPlayers; }
    public int getTotalRounds() { return totalRounds; }
    public String getFirstRoundStartTime() { return firstRoundStartTime; }
    public List<PlayerInfo> getAllPlayers() { return allPlayers; }
}
